globus_cli/config.py

```diff
@@ -87,8 +86,14 @@ def get_config_obj(system=False):
 
 
 def lookup_option(option, section='cli', environment=None):
-    p = globus_sdk.config._get_parser()
-    return p.get(option, section=section, environment=environment)
+    conf = get_config_obj()
+    try:
+        if environment:
+            return conf["environment " + environment][option]
+        else:
+            return conf[section][option]
+    except KeyError:
+        return None
```

Comment:
You said this is the part that makes you nervous. I think this change is okay, and probably for the better long-term.

It _is_ a subtle behavioral change, and could break some esoteric setups, but it's acceptable.
The SDK config parser uses the standard python `ConfigParser`, and reads from `~/.globus.cfg` and `/etc/globus.cfg`. The CLI config obj is from `configobj`, which, because it thinks of config as something which can be flushed back to disk, can't handle multiple files.

So, if someone had something in `/etc/globus.cfg` which they were relying on in the CLI, it would break.
No one normally will, and I'm okay with breaking those who do -- they'll be primarily internal if they even exist.